### ISSUE TITLE
loclist jump doesn't work for :GoFmt

### ISSUE THREAD

**Issue by alaska** [fatih/vim-go#1182]
### Behavior

Sometime during the last couple of releases (working to bisect), the errors listed in the location list from gofmt no longer jump to the line where the error is listed.  <CR> does nothing

### Steps to reproduce:

1. Open any .go file
2. Add a random number sign `#` - `U+0023` to an empty line
3. Save

### Expected results
- Loclist opens with output like the following:
`|184 col 2| : illegal character U+0023 '#'`
- hitting enter `<CR>` jumps to the line in the buffer you came from

### Actual results
- Loclist appears as above
- hitting enter `<CR>` has no visible effect

### Configuration

All other plugins disabled

### vimrc
```
set nocompatible
filetype off
set rtp+=~/.vim/bundle/Vundle.vim
call vundle#begin()
Plugin 'VundleVim/Vundle.vim'
Plugin 'fatih/vim-go'
call vundle#end()
filetype plugin indent on
let g:go_template_use_pkg = 1

"disable arrows
noremap <up> <nop>
inoremap <up> <nop>
noremap <down> <nop>
inoremap <down> <nop>
noremap <left> <nop>
inoremap <left> <nop>
noremap <right> <nop>
inoremap <right> <nop>

syntax on

" Colors and theme
set t_Co=256
set background=light
let g:nofrils_strbackgrounds=1
colorscheme nofrils-light

" Set up <TAB> length and behavior
set tabstop=4
set shiftwidth=4
set expandtab
" This makes backspace treat four spaces like a tab:
set softtabstop=4
set shiftround
```

### version
```
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Jul 28 2016 14:11:47)
MacOS X (unix) version
Included patches: 1-2109
Compiled by Homebrew
Huge version without GUI.  Features included (+) or not (-):
+acl             +channel         +comments        +diff            +extra_search    +fork()          +keymap          -lua             -mouse_gpm       +multi_byte      +perl            +quickfix        +startuptime     -tcl             -toolbar         +vreplace        -xim
+arabic          +cindent         +conceal         +digraphs        +farsi           -gettext         +langmap         +menu            -mouse_jsbterm   +multi_lang      +persistent_undo +reltime         +statusline      +termguicolors   +user_commands   +wildignore      -xsmp
+autocmd         -clientserver    +cryptv          -dnd             +file_in_path    -hangul_input    +libcall         +mksession       +mouse_netterm   -mzscheme        +postscript      +rightleft       -sun_workshop    +terminfo        +vertsplit       +wildmenu        -xterm_clipboard
-balloon_eval    +clipboard       +cscope          -ebcdic          +find_in_path    +iconv           +linebreak       +modify_fname    +mouse_sgr       +netbeans_intg   +printer         +ruby            +syntax          +termresponse    +virtualedit     +windows         -xterm_save
-browse          +cmdline_compl   +cursorbind      +emacs_tags      +float           +insert_expand   +lispindent      +mouse           -mouse_sysmouse  +num64           +profile         +scrollbind      +tag_binary      +textobjects     +visual          +writebackup     -xpm
++builtin_terms  +cmdline_hist    +cursorshape     +eval            +folding         +job             +listcmds        -mouseshape      +mouse_urxvt     +packages        +python          +signs           +tag_old_static  +timers          +visualextra     -X11
+byte_offset     +cmdline_info    +dialog_con      +ex_extra        -footer          +jumplist        +localmap        +mouse_dec       +mouse_xterm     +path_extra      -python3         +smartindent     -tag_any_white   +title           +viminfo         -xfontset
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: clang -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X_UNIX  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: clang   -L. -fstack-protector -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib  -L/usr/local/lib -o vim        -lncurses -liconv -framework Cocoa   -mmacosx-version-min=10.11 -fstack-protector-strong -L/usr/local/lib  -L/usr/local/Cellar/perl/5.24.0_1/l
ib/perl5/5.24.0/darwin-thread-multi-2level/CORE -lperl -lm -lutil -lc -F/usr/local/Cellar/python/2.7.12/Frameworks -framework Python   -lruby.2.3.0 -lobjc -L/usr/local/Cellar/ruby/2.3.1/lib
```

### other data
* vim-go version: `1.11`
* go version: `g1.7.4 darwin/amd64`

**Comment by tarrant** [comment]
It appears that the issue is caused by the location list no longer containing the file name. In older versions where this works you will get a location list along the lines of:

```
foobar/foo.go|39 col 1| : expected declaration, found '}'
```

While now instead it will return:
```
|39 col 1| : expected declaration, found '}'
```

I have a patch that seems to fix the issue. I'll submit it shortly.

**Comment by fatih** [comment]
Fixed with #1199. Please let me know if this doesnt work.
